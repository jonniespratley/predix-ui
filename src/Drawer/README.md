This is the documentation.

- The drawer by default is hidden off screen and only visible when open is `true`.
- Setting the `docked` prop to `true` will make the overlay hidden and drawer visible.



## Usage



```react

state:
  overlay: false
  docked: false
  clicked: 0
  open: false
---
const style = {
  height: 300,
  overflow: 'hidden',
  position: 'relative'
};
const { Drawer } = PxReact;
<div>
  <Drawer opened={true} anchor='left'>
    <p>This is drawer content</p>
  </Drawer>
</div>
```


### Temporary drawer
Temporary navigation drawers can toggle open or closed. Closed by default, the drawer opens temporarily above all other content until a section is selected.


```
// code here
```

### Permanent drawer
Permanent navigation drawers are always visible and pinned to the left edge, at the same elevation as the content or background. They cannot be closed.

```
// code here
```


### Persistent drawer
Persistent navigation drawers can toggle open or closed. The drawer sits on the same surface elevation as the content. It is closed by default and opens by selecting the menu icon, and stays open until closed by the user. The state of the drawer is remembered from action to action and session to session.

When the drawer is outside of the page grid and opens, the drawer forces other content to change size and adapt to the smaller viewport.

```
// code here
```


### Mini drawer


```
// code here
```


## Properties

```table
span: 6
rows:
  - Name: children
    Type: String
    Description: The contents of the Drawer.

  - Name: docked
    Type: bool
    Description: If true, the drawer will be docked, the overlay won't show and clicking items will not close the drawer.

  - Name: open
    Type: bool
    Description: If true, the Drawer is opened. Providing a value will turn the Drawer into a controlled component.

  - Name: overlayClassName
    Type: String
    Description: The CSS class name to add to the Overlay component that is rendered behind the Drawer

  - Name: overlayStyle
    Type: object
    Description: Override the inline-styles of the Overlay component that is rendered behind the Drawer.
```




## Styling

Custom property	                    Description	Default

```
--px-drawer	                 Style mixin to be applied to the element	{}
--px-drawer-content	          Style mixin to be applied to the content	{}
--px-drawer-opened	           Style mixin to be applied to the element when opened	{}
--px-drawer-background-color	   The background-color of the drawer	#242326 (gray10)
--px-drawer-width	                 The width of the drawer	256px
--px-drawer-height	                The height of the drawer	100vh
--px-drawer-overlay-background	    The background color of the overlay	rgba(0, 0, 0, 0.5))
```

 