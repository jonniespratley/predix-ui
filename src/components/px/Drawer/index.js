import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

// Drawer
const Drawer = styled.div`
  position: fixed;
  z-index: var(--px-drawer-z-index, 100);
  color: var(--px-drawer-font-color, black);
  background-color: var(--px-drawer-background-color, white);
  width: var(--px-drawer-width, 256px);
  height: var(--px-drawer-height, 100%);
  box-shadow: var(--px-drawer-box-shadow, rgba(0, 0, 0, 0.16) 0 1px 4px, rgba(0, 0, 0, 0.23) 0 1px 4px);

  transition: var(--px-drawer-transition, transform 350ms cubic-bezier(0.23, 1, 0.32, 1));

  ${props => props.anchor === 'left' && css`
    top: 0;
    left: 0;
    right: auto;
    transform: translateX(-110%);
    ${props.opened && css`
      transform: translateX(0);
    `}
  `}

  ${props => props.anchor === 'right' && css`
    top: 0;
    left: auto;
    right: 0;
    transform: translateX(110%);
    ${props.opened && css`
      transform: translateX(0);
    `}
  `}

  ${props => props.anchor === 'top' && css`
    top: 0;
    left: 0;
    bottom: auto;
    right: 0;
    height: auto;
    max-height: 100vh;
    transform: translateY(-110%);
    width: 100%;
    ${props.opened && css`
      transform: translateY(0);
    `}
  `}

  ${props => props.anchor === 'bottom' && css`
    top: auto;
    left: 0;
    bottom: 0;
    right: 0;
    height: auto;
    max-height: 100vh;
    width: 100%;
    transform: translateY(105%);
    width: 100%;
    ${props.opened && css`
      transform: translateY(0);
    `}
  `}

  &::after {
    position: fixed;
    top: 56px;
    bottom: 0;
    left: 100%;
    visibility: visible;
    width: 20px;
    content: "";
  }

  ${props => props.opened && css`
    z-index: var(--px-drawer-z-index, 25);
    visibility: visible;
  `}

  ${props => props.docked && css`
    z-index: var(--px-drawer-z-index, 25);
    visibility: visible;
    transform: translateX(0);
    flex: 0 0 auto;
  `}
`;
Drawer.displayName = 'Drawer';
Drawer.defaultProps = {
  className: 'px-drawer'
};

// Content
const DrawerContent = styled.div`
  overflow-x: hidden;
  overflow-y: auto;
  display: flex;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  flex: 1 0 auto;
`;
DrawerContent.displayName = 'DrawerContent';
DrawerContent.defaultProps = {
  className: 'px-drawer-content'
};

// Overlay
const DrawerOverlay = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  transition: opacity 0.2s ease;
  transform: translateZ(0);
  opacity: 0;
  background: var(--px-drawer-overlay-background, rgba(0, 0, 0, 0.5));
  z-index: -1;
  visibility: hidden;

  ${props => props.opened && css`
    opacity: 1;
    z-index: 15;
    visibility: visible;
    display: block;
  `}
`;
DrawerOverlay.displayName = 'DrawerOverlay';
DrawerOverlay.defaultProps = {
  className: 'px-drawer-overlay'
};


const DrawerComponent = ({
  fixed,
  overlay,
  onOverlayClick,
  opened,
  docked,
  type,
  anchor,
  children
}) => (
  <div>
    <Drawer opened={opened} anchor={anchor} fixed={fixed} docked={docked} type={type}>
      <DrawerContent>{children}</DrawerContent>
    </Drawer>
    {overlay && <DrawerOverlay onClick={onOverlayClick} opened={opened} />}
  </div>
);

DrawerComponent.propTypes = {
  opened: PropTypes.bool,
  fixed: PropTypes.bool,
  docked: PropTypes.bool,
  overlay: PropTypes.bool,
  anchor: PropTypes.string,
  onOverlayClick: PropTypes.func,
  type: PropTypes.string,
  children: PropTypes.node
};

DrawerComponent.defaultProps = {
  opened: false,
  fixed: false,
  overlay: false,
  docked: false,
  onOverlayClick: null,
  type: null,
  children: null,
  anchor: 'left'
};

export default DrawerComponent;
