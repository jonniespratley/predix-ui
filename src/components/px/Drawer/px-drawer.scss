@import "px-mobile-design/sass/_mixins.scss";


$drawer-transition: transform 450ms cubic-bezier(0.23, 1, 0.32, 1);
$drawer-overlay-background-color: rgba(0, 0, 0, 0.5);
$drawer-background-color: white;
$drawer-font-color: black;

$drawer-box-shadow: rgba(0, 0, 0, 0.16) 0 1px 4px, rgba(0, 0, 0, 0.23) 0 1px 4px;

$drawer-height: 100%;
$drawer-width: 256px;
$drawer-max-width: 300px;

/**
 * Drawer styles
 */
.px-drawer {

  	///------------------------------------------------------
  	/// default
  	.drawer {
  		transition: $drawer-transition;
  		z-index: 20;
  		color: var(--px-drawer-font-color, $drawer-font-color);
  		background-color: var(--px-drawer-background-color, $drawer-background-color);
  		width: var(--px-drawer-width, $drawer-width);
  		height: var(--px-drawer-height, $drawer-height);
  		box-shadow: $drawer-box-shadow;
      position: fixed;
      transform: translateX(-110%);
      //visibility: hidden;
      top: 0;
      left: 0;
      bottom: 0;
  		&::after {
  			position: fixed;
  			top: 56px;
  			bottom: 0;
  			left: 100%;
  			visibility: visible;
  			width: 20px;
  			content: "";
  		}
      &--is-fixed {
        position: fixed;
      }

  		&--is-docked {

  			transform: translateX(0);
  		}

  		&--is-opened{
  			visibility: visible;
  			transform: translateX(0);
  		}


  		&__content {
  			overflow-x: hidden;
  		}


  		&__overlay {
  			position: fixed;
  			top: 0;
  			right: 0;
  			bottom: 0;
  			left: 0;
  			transition: opacity 0.2s ease;
  			transform: translateZ(0);
  			opacity: 0;
  			background: var(
  				--px-drawer-overlay-background,
  				$drawer-overlay-background-color
  			);
  			z-index: -1;
  			visibility: hidden;
  			display: none;

  			&--is-open {
  				opacity: 1;
  				z-index: 15;
  				visibility: visible;
  				display: block;
  			}
  		}

  	}
}
