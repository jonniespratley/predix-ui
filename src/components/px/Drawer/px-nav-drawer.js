import React from 'react';
import BaseComponent from '../BaseComponent';

import style from './px-nav-drawer.scss';
import classnames from 'classnames';

class NavDrawer extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      opened: props.opened || false,
      persistent: props.persistent || false
    };
    this.close = this.close.bind(this);
  }

	componentDidMount() {
		this.hasUnprefixedTransform = 'transform' in document.documentElement.style;
		//this.rootElement = this.base.querySelector('.js-nav-drawer');
		//this.sideNavContent = this.refs.rootElement.querySelector('.js-nav-drawer-content');
		//this.sideNavBody = this.refs.rootElement.querySelector('.nav-drawer__body');
		this.refs.sideNavContent.addEventListener('click', (e) => {
			e.stopPropagation();
		});
		if (this.hasUnprefixedTransform) {
			this._setupTouchHandlers();
		}
	}

  componentWillReceiveProps(nextProps){
   // console.log('NavDrawer.componentWillReceiveProps', nextProps);

    //this.setState(nextProps);
    //this.toggle();
  }

  getWidth(){
    return this.refs.sideNavContent.offsetWidth;
  }

	_setupTouchHandlers() {
		this.touchStartX = null;
		this.sideNavTransform = null;
		this.refs.sideNavContent.addEventListener('touchstart', this.onSideNavTouchStart.bind(this));
		this.refs.sideNavContent.addEventListener('touchmove', this.onSideNavTouchMove.bind(this));
		this.refs.sideNavContent.addEventListener('touchend', this.onSideNavTouchEnd.bind(this));
	}

	onSideNavTouchStart(e) {
		e.preventDefault();
		this.touchStartX = e.touches[0].pageX;
	}

	onSideNavTouchMove(e) {
		e.preventDefault();
		const newTouchX = e.touches[0].pageX;
		this.sideNavTransform = Math.min(0, newTouchX - this.touchStartX);
		this.refs.sideNavContent.style.transform = 'translateX(' + this.sideNavTransform + 'px)';
	}

	onSideNavTouchEnd(e) {
		let TOUCH_SLOP = 8 * window.devicePixelRatio * 3;
		if (this.sideNavTransform < -TOUCH_SLOP) {
			this.close();
			return;
		}
		this.open();
	}

	isOpen() {
		return this.refs.rootElement.classList.contains('nav-drawer--visible');
    //this.props.opened;
	}

	toggle() {
		if (this.isOpen()) {
			this.close();
		} else {
			this.open();
		}
	}

	close() {
    //console.log('px-nav-drawer', 'close', this);
    //this.setState({opened: false});
		this.refs.baseElement.classList.remove('px-nav-drawer--is-visible');
		this.refs.rootElement.classList.remove('nav-drawer--visible');
		this.refs.sideNavContent.classList.add('nav-drawer__content--animatable');
		if (this.hasUnprefixedTransform) {
			this.refs.sideNavContent.style.transform = 'translateX(-102%)';
		} else {
			this.refs.sideNavContent.classList.remove('nav-drawer--visible');
		}
    if(this.props.onClose){
      this.props.onClose();
    }


    return false;
	}

	open() {
    //this.setState({opened: true});
		this.refs.rootElement.classList.add('nav-drawer--visible');
		if (this.hasUnprefixedTransform) {
			let onSideNavTransitionEnd = (e) => {
				this.refs.sideNavBody.focus();
				this.refs.sideNavContent.classList.remove('nav-drawer__content--animatable');
				this.refs.sideNavContent.removeEventListener('transitionend', onSideNavTransitionEnd);
			};

			this.refs.sideNavContent.classList.add('nav-drawer__content--animatable');
			this.refs.sideNavContent.addEventListener('transitionend', onSideNavTransitionEnd);
			requestAnimationFrame(() => {
				this.refs.sideNavContent.style.transform = 'translateX(0px)';
			});
		} else {
			this.refs.sideNavContent.classList.add('nav-drawer--visible');
		}
    if(this.props.onOpen){
      this.props.onOpen();
    }
   // console.log('px-nav-drawer', 'open', this);
	}

	render() {
    const {
  		title = 'AppNav',
  		nav,
      opened,
      persistent,
      children
  	} = this.props;

    const baseClasses = classnames(
      'px-nav-drawer',
      {'px-nav-drawer--is-visible': opened},
      {'px-nav-drawer--is-persistent': persistent}
    );

    const rootElementClasses = classnames(
      'nav-drawer',
      'js-nav-drawer',
    //  {'nav-drawer--visible': opened},
      {'nav-drawer--persistent': persistent}
    );

    const sideNavContentClasses = classnames(
      'nav-drawer__content',
      'js-nav-drawer-content'
    );


		return (
			<div className={baseClasses} ref='baseElement'>
				<section className={rootElementClasses} ref='rootElement' onClick={this.close}>
					<div className={sideNavContentClasses} ref='sideNavContent' onClick={this.close}>
						<div className="nav-drawer__header">
							<h1 className="nav-drawer__title">{title}</h1>
						</div>
						<nav className="nav-drawer__body" ref='sideNavBody'>
              {children}
							{nav && nav.map((item) => (
								<a href={item.url} role="tab" activeClassName="active" className="nav-drawer__link">
									<span onClick={(e) => this.close(e)}>{item.label}</span>
								</a>
							))}
						</nav>
					</div>
				</section>
        <style jsx>{style}</style>
			</div>
		);
	}
}

NavDrawer.defaultProps = {
  opened: false,
  persistent: false
};
export default NavDrawer;
