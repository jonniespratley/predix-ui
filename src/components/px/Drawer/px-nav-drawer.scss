$global-nav-background-color: black;
$global-background-color: #222;
$global-nav-font-color: white;

.px-nav-drawer{


  .nav-drawer {
    width         : 100%;
    height        : 100%;
    position      : fixed;
    pointer-events: none;
    top           : 0;
    left          : 0;
    overflow      : hidden;
  }

  .nav-drawer:before {
    content    : '';
    width      : 100%;
    height     : 100%;
    background : var(--px-nav-drawer-background-color, black);
    opacity    : 0;
    display    : block;
    position   : absolute;
    will-change: opacity;
    transition : opacity 0.233s cubic-bezier(0,0,0.21,1);
  }

  .nav-drawer--visible {
    pointer-events: auto;
    z-index: 100;
  }

  .nav-drawer--visible:before {
    opacity: 0.7;
  }

  .nav-drawer__content {
    background : var(--px-nav-drawer-content-background-color, white);
    width      : 80%;
    max-width  : 304px;
    height     : 100%;
    overflow   : hidden;
    position   : relative;
    box-shadow : 0 0 4px rgba(0, 0, 0, .14), 0 4px 8px rgba(0, 0, 0, .28);
    will-change: transform;
    transform  : translateX(-102%);
    &.nav-drawer--visible {
      transform: translateX(0px);
      z-index: auto;
    }
  }

  .nav-drawer__content--animatable {
    transition: transform 0.233s cubic-bezier(0,0,0.21,1);
  }

  .nav-drawer__header {
    background     : var(--px-nav-drawer-header-background-color, black);
    background-size: cover;
    width          : 100%;
    height         : 75px;
    position       : relative;
    user-select: none;
  }

  .nav-drawer__title {
    font-size  : 16px;
    line-height: 1;
    color      : $global-nav-font-color;
    position   : absolute;
    bottom     : 8px;
    left       : 16px;
    height     : 16px;
    font-weight: 500;
  }

  .nav-drawer__body {
    padding-top: 8px;
  }

  .nav-drawer__version {
    position : absolute;
    bottom   : 16px;
    left     : 16px;
    font-size: 14px;
    opacity  : 0.54;
  }

  .nav-drawer__delete-all,
  .nav-drawer__delete-memos,
  .nav-drawer__file-bug-report,
  .nav-drawer__link {
    font-size    : 14px;
    outline      : none;
    height       : 48px;
    padding-left : 20px;
    width        : 100%;
    text-align   : left;
    display      : block;
    border       : none;
    //background: url(/images/ic_delete_24px.svg) 16px 12px no-repeat;
    color        : rgba(0,0,0,0.87);
    cursor       : pointer;
    display      : flex;
    align-content: stretch;

    span {
      flex: 1;
    }

    &:hover {
      background-color: rgba(51, 51, 51, 0.21);
    }
  }

  .nav-drawer__delete-all {
    //background-image: url(/images/ic_restore_24px.svg);
  }

  .nav-drawer__link {
    //background-image: url(/images/ic_info_outline_24px.svg);
    line-height    : 48px;
    text-decoration: none;
  }

  .nav-drawer__link:focus {
    background-color: #eee;
    outline         : 0;
  }

  .nav-drawer__file-bug-report {
    //  background-image: url(/images/ic_feedback_24px.svg);
    line-height    : 48px;
    text-decoration: none;
  }

  .nav-drawer__link.active {
    font-weight     : bold;
    background-color: rgba(0,0,0,.05);
  }



    &--is-visible{
      background: red;
      .nav-drawer{
        pointer-events: auto;
        z-index: 100;
      }
      .nav-drawer:before{
        opacity: 0.7;
      }
      .nav-drawer__content {
        transform: translateX(0px);
        z-index: auto;
      }
    }

    &--is-persistent{
      background: blue;
      .nav-drawer{
        position: absolute !important;
      }
    }
}
