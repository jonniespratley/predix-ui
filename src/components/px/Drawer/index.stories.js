import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, select } from '@storybook/addon-knobs';
import Drawer from './';
import Navbar from '../Navbar';

const style = {
  height: 300,
  overflow: 'hidden',
  position: 'relative'
};

const anchorOptions = [
  'top',
  'bottom',
  'left',
  'right'
];

storiesOf('Components / Drawer', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <Drawer
      overlay={boolean('overlay', false)}
      onClose={action('onClose')}
      onOpen={action('onOpen')}
      onOverlayClick={action('onOverlayClick')}
      opened={boolean('opened', false)}
      docked={boolean('docked', false)}
      anchor={select('anchor', anchorOptions, 'left')}
    >
      <p>This is the content of the drawer</p>
    </Drawer>
  ))
  .add('with layout', () => (
    <div>
      <div style={style}>
        <Navbar
          title="Navbar"
          subtitle="Subtitle"
          onMenuButtonClick={action('onMenuButtonClick')}
          showMenuButton
        />

        <Drawer
          overlay={boolean('overlay', false)}
          onClose={action('onClose')}
          onOpen={action('onOpen')}
          onOverlayClick={action('onOverlayClick')}
          opened={boolean('opened', false)}
          docked={boolean('docked', false)}
          anchor={select('anchor', anchorOptions, 'left')}
        >
          <p>This is the content of the drawer</p>
        </Drawer>
        <div className="content u-p">
          <p>This is content</p>
        </div>

      </div>
    </div>
  ));
