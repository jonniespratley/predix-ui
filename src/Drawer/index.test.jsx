import React from 'react';
import renderer from 'react-test-renderer';

import Drawer from './Drawer';

describe('Drawer', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<Drawer id="d1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('opened - matches snapshot', () => {
    const tree = renderer.create(<Drawer id="d1" opened />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('docked - matches snapshot', () => {
    const tree = renderer.create(<Drawer id="d1" docked />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  ['top', 'bottom', 'left', 'right'].forEach((anchor) => {
    test(`${anchor} - matches snapshot`, () => {
      const tree = renderer.create(<Drawer id="d1" anchor={anchor} opened />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
