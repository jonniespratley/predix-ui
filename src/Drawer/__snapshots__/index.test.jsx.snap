// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer bottom - matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: auto;
  left: 0;
  bottom: 0;
  right: 0;
  height: auto;
  max-height: 100vh;
  width: 100%;
  -webkit-transform: translateY(105%);
  -ms-transform: translateY(105%);
  transform: translateY(105%);
  width: 100%;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  z-index: var(--px-drawer-z-index,25);
  visibility: visible;
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;

exports[`Drawer docked - matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: 0;
  left: 0;
  right: auto;
  -webkit-transform: translateX(-110%);
  -ms-transform: translateX(-110%);
  transform: translateX(-110%);
  z-index: var(--px-drawer-z-index,25);
  visibility: visible;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;

exports[`Drawer left - matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: 0;
  left: 0;
  right: auto;
  -webkit-transform: translateX(-110%);
  -ms-transform: translateX(-110%);
  transform: translateX(-110%);
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  z-index: var(--px-drawer-z-index,25);
  visibility: visible;
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;

exports[`Drawer matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: 0;
  left: 0;
  right: auto;
  -webkit-transform: translateX(-110%);
  -ms-transform: translateX(-110%);
  transform: translateX(-110%);
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;

exports[`Drawer opened - matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: 0;
  left: 0;
  right: auto;
  -webkit-transform: translateX(-110%);
  -ms-transform: translateX(-110%);
  transform: translateX(-110%);
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  z-index: var(--px-drawer-z-index,25);
  visibility: visible;
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;

exports[`Drawer right - matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: 0;
  left: auto;
  right: 0;
  -webkit-transform: translateX(110%);
  -ms-transform: translateX(110%);
  transform: translateX(110%);
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  z-index: var(--px-drawer-z-index,25);
  visibility: visible;
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;

exports[`Drawer top - matches snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: var(--px-drawer-z-index,100);
  color: var(--px-drawer-font-color,black);
  background-color: var(--px-drawer-background-color,white);
  width: var(--px-drawer-width,256px);
  height: var(--px-drawer-height,100%);
  box-shadow: var(--px-drawer-box-shadow,rgba(0,0,0,0.16) 0 1px 4px,rgba(0,0,0,0.23) 0 1px 4px);
  -webkit-transition: var(--px-drawer-transition,-webkit-transform 350ms cubic-bezier(0.23,1,0.32,1));
  -webkit-transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  transition: var(--px-drawer-transition,transform 350ms cubic-bezier(0.23,1,0.32,1));
  top: 0;
  left: 0;
  bottom: auto;
  right: 0;
  height: auto;
  max-height: 100vh;
  -webkit-transform: translateY(-110%);
  -ms-transform: translateY(-110%);
  transform: translateY(-110%);
  width: 100%;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  z-index: var(--px-drawer-z-index,25);
  visibility: visible;
}

.c0::after {
  position: fixed;
  top: 56px;
  bottom: 0;
  left: 100%;
  visibility: visible;
  width: 20px;
  content: "";
}

.c1 {
  overflow-x: hidden;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-overflow-scrolling: touch;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

<div>
  <div
    className="c0 px-drawer"
    type={null}
  >
    <div
      className="c1 px-drawer-content"
    />
  </div>
</div>
`;
